operator/pkg/validate/validate.go

```diff
@@ -200,6 +201,31 @@ func validateRevision(_ util.Path, val interface{}) util.Errors {
 	return nil
 }
 
+// IsRevisionString returns error if non-string revision is passed
+func IsRevisionString(iopYAML string) error {
+	var specTree = make(map[string]interface{})
+	spec, err := tpath.GetConfigSubtree(iopYAML, "spec")
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal([]byte(spec), &specTree)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling spec overlay yaml into untype tree %v", err)
```

Comment:
This message will appear to users with invalid YAML, and it will make no sense to them.  Use a more generic message "invalid IstioOperator: %v" or change the wording to something that will make sense to someone who doesn't know Istio internals.

Reply:
Done.